docs/datagrid/performance.md

```diff
@@ -0,0 +1,57 @@
+# Performance and rendering
+Row virtualization is a performance optimization technique used in DataGrid.  Instead of rendering all the rows of data at once, which can be thousands of entries, row virtualization renders only the rows that are currently visible within the user's viewport (plus a small buffer). As the user scrolls, the DataGrid dynamically renders new rows and removes those that scroll out of view.
+
+![Illustration showing virtualization of rows](ill_virtualization.png)
+
+This technique significantly enhances DataGrid performance, leading to faster load times and efficient updates. Row virtualization reduces memory consumption, allowing the application to handle large datasets smoothly without compromising performance. Users benefit from a smoother experience with fluid scrolling and more responsive interactions when e.g. sorting rows.
+
+## Optimizing performance
+```js
+{
+    rendering: {
+        rows: {
+            bufferSize: 3, // default is 10
+            strictHeights: true // default is false
+        }
+    }
+}
+```
+
+The `bufferSize` and `strictHeights` options can be adjusted to optimize performance and smoothness of scrolling based on your spesific use case and preferences. 
+
+### buffersize
+Defines the number of rows rendered outside the viewport (the buffer) during scrolling. A small buffer leads to faster initial rendering, and increased performance on sorting and scroll. A larger buffer means slower initial rendering, but might decrease flicker effect on fast scrolling.
+
+The default `bufferSize` of 10 should be optimal in most cases. 
+
+### strictHeights
+By default, rows adjust their height to fit all content, which can reduce performance and scrolling smoothness. Setting `strictHeights: true` ensures all rows have a uniform height, truncating multiline text with an ellipsis. This skips height calculations and boosts performance.
```

Comment:
It would be good to add a link to API docs.

Reply:
@sebastianbochan: agree, but I don't know how to find correct links to API ref that is not in production. Could you help out here and "Add a suggestion"?